libnavigation-base/src/main/java/com/mapbox/navigation/base/route/OnboardRouter.kt

```diff
@@ -0,0 +1,3 @@
+package com.mapbox.navigation.base.route
+
+interface OnboardRouter : Router
```

Comment:
why do we need separate _interface_ for each type of `Router`?

Reply:
I was testing out a Java-type based system. This didn't pan out so reverting back to enum-types.